airbyte-webapp/src/components/JobItem/components/Logs.tsx

```diff
@@ -57,4 +61,41 @@ const Logs: React.FC<LogsProps> = ({ logsArray }) => {
   );
 };
 
+/**
+ * Matching the log's line number by time makes the following assumptions:
+ * 1. The log's lines are already ordered by time
+ * 2. The timestamps used are in the same timezone
+ * 3. The error is closer to the end of the log file than the beginning
+ * 4. Lines are matched by the start of a line with "YYYY-MM-DD HH:mm:ss" format, like "2022-05-17 23:00:19 DEBUG I am a log message"
+ */
+const getMatchingLineNumbers = (matchTimestamp: number | undefined, lines: string[] | undefined) => {
+  if (!matchTimestamp || !lines || lines.length === 0) {
+    return [];
+  }
+  const flooredMatchTimestamp = Math.floor(matchTimestamp / 1000) * 1000;
+
+  const resolutionOffset = 1000; // the resolution of the timestamps is in seconds
```

Comment:
:question: I am not fully udnerstanding why we need this? Why are we expecting the `failure.timestamp` to be not exactly the one in the log, that we try to search a bit "around" that time?

Reply:
Exactly - the Platform stores the timestamp of the failure with ms resolution... but the log lines we get back are in second resolution.  So the failure may have occurred within +/- 1 second for the logged timestamp.

Reply:
I see, thanks makes sense for me now. Wouldn't it thought be easier (and potentially safer, so we're not getting "too early" log lines), to just conver the `failure.timestamp` into a second timestamp here, and compare against those matching? Otherwise we might match also on log timestamps that are in the second before the error?

Reply:
Moved to matching time in full 1-second chunks via 53a9f51f5cd8030caeb2b0415ad0da56e4a1d8ec